packages/buckets/src/buckets.ts

```diff
@@ -225,37 +240,28 @@ export class Buckets {
    * }
    * ````
    */
-  async list(ctx?: ContextInterface) {
+  async list() {
     logger.debug('list request')
-    const req = new pb.ListRequest()
-    const res: pb.ListReply = await this.unary(API.List, req, ctx)
-    return res.toObject().rootsList
+    return bucketsList(this)
   }
 
   /**
    * Returns information about a bucket path.
    * @param key Unique (IPNS compatible) identifier key for a bucket.
    * @param path A file/object (sub)-path within a bucket.
    */
-  async listPath(key: string, path: string, ctx?: ContextInterface) {
+  async listPath(key: string, path: string): Promise<ListPathReply.AsObject> {
     logger.debug('list path request')
-    const req = new pb.ListPathRequest()
-    req.setKey(key)
-    req.setPath(path)
-    const res: pb.ListPathReply = await this.unary(API.ListPath, req, ctx)
-    return res.toObject()
+    return bucketsListPath(this, key, path)
```

Comment:
here's an example using the function method but adding it here to the stateful class so that documentation will be clear